src/persist/src/postgres.rs

```diff
@@ -138,7 +139,9 @@ impl PostgresConsensus {
         // TODO: reconsider opening with NoTLS. Perhaps we actually want to,
         // especially given the fact that its not entirely known what data
         // will actually be stored in Consensus.
-        let (mut client, conn) = tokio_postgres::connect(&config.url, NoTls).await?;
+        let (mut client, conn) = tokio_postgres::connect(&config.url, NoTls)
+            .await
+            .with_context(|| format!("error connecting to postgres at '{}'", config.url))?;
```

Comment:
Ah yeah, would be best not to include `config.url` in the error message since as @danhhz points out that might include a password:
```suggestion
            .with_context(|| "error connecting to postgres")?;
```

Reply:
done